search_indices/__init__.py

```diff
@@ -28,8 +33,14 @@ def get_raudikko_filter() -> token_filter:
 
 
 def get_finnish_analyzer() -> analyzer:
-    es_connection = create_elasticsearch_connection()
-    if "raudikko" in es_connection.cat.plugins():
+    try:
+        es_connection = create_elasticsearch_connection()
+        es_plugins = es_connection.cat.plugins()
+    except elasticsearch.ConnectionError:
```

Comment:
Looks like all the calls to get_finnish_analyzer() store the result on a module global variable.. so there's a tiny but non-zero chance of ending up with a uwsgi worker process in prod that configures itself to use the raudikko default alternative due to some temporary connection issue.

So in theory could end up in a fun :tm:  situation where workers in one process return different results from workers in another process due to having different analyzer configuration. Again, quite unlikely, but not impossible. Or am I missing something?

This was done because of the collectstatic failing in docker build, right? So okay.. uh, first thought was to make this function evaluate lazily, but on a closer look that's not gonna help since the results are evaluated immediately, at import time, by the function callers.

IDK, should this whole thing be rather configured by the environment rather than be always runtime probed? If we want to keep it backwards compatible, it could have three modes set by an environment variable: probe (default), raudikko and standard. We can then just set our variable to raudikko since that's what we're using.

Reply:
Like I'm thinking something like this in settings

```python
ELASTICSEARCH_ANALYZER_MODE = probe | raudikko | standard

raudikko_finnish_analyzer = analyzer(...)
standard_finnish_analyzer = analyzer(...)

def get_finnish_analyzer(mode):
    if mode == "probe":
        # check ES plugins via ES connection
    elif mode == "raudikko":
        return raudikko_finnish_analyzer

FINNISH_ANALYZER = get_finnish_analyzer(ELASTICSEARCH_ANALYZER_MODE)
```

and then just refer to settings.FINNISH_ANALYZER where we have currently `finnish_analyzer = get_finnish_analyzer()`

Reply:
Indeed there's a small non-zero chance of things going sideways. I'll do some refactoring and I'll consider your advice. :+1: